src/main/scala/akka/contrib/datareplication/ORMultiMap.scala

```diff
@@ -75,26 +75,17 @@ case class ORMultiMap private (private[akka] val map: ORMap)
    * Associate an entire set with the key while retaining the history of the previous
    * replicated data set.
    */
-  def put(node: Cluster, key: String, value: Set[Any]): ORMultiMap =
+  def put[T](node: Cluster, key: String, value: Set[T]): ORMultiMap =
     put(node.selfUniqueAddress, key, value)
 
-  /*
-   * Remove all elements of a set and then merge in elements of a new set so that we may retain the replicated
-   * data history throughout.
-   * TODO: We may wish to consider optimising this as the code isn't the most efficient. Ideally we should be able to remove all entries of an ORSet in one go. The same goes for adding to an ORSet in bulk.
-   */
-  private def removeAndMerge(node: UniqueAddress, newValue: ORSet)(oldValue: ORSet): ORSet =
-    oldValue.value.diff(newValue.value)
-      .foldLeft(oldValue) { (value, element) ⇒ value.remove(node, element) }
-      .merge(newValue)
-
   /**
    * INTERNAL API
    */
-  private[akka] def put(node: UniqueAddress, key: String, value: Set[Any]): ORMultiMap = {
-    val newValue = value.foldLeft(ORSet.empty)((v, e) ⇒ v.add(node, e))
-    val values = updateOrInit(key, removeAndMerge(node, newValue), newValue)
-    ORMultiMap(map.put(node, key, values))
+  private[akka] def put[T](node: UniqueAddress, key: String, value: Set[T]): ORMultiMap = {
+    val newUnderlying = underlying.updated(node, key, ORSet.empty) { existing =>
+      value.foldLeft(existing.clear(node)) { (s, element) => s.add(node, element) }
+    }
+    ORMultiMap(newUnderlying)
```

Comment:
@huntc I have adjusted the ORMultiMap to use the new `ORMap.updated`, which is very similar to you what you had in `updateOrInit`